llama-index-integrations/vector_stores/llama-index-vector-stores-lancedb/llama_index/vector_stores/lancedb/base.py

```diff
@@ -112,10 +118,16 @@ def __init__(
         refine_factor: Optional[int] = None,
         text_key: str = DEFAULT_TEXT_KEY,
         doc_id_key: str = DEFAULT_DOC_ID_KEY,
+        api_key: Optional[str] = None,
+        region: Optional[str] = None,
         **kwargs: Any,
     ) -> None:
         """Init params."""
-        self._connection = lancedb.connect(uri)
+        if api_key is not None:
```

Comment:
The conditions that we should we addressing are:
* if uri is remote but api_key is not provided - raise error
* if api_key is provided with local uri - Just warn that "api key provided with local uri. The data will be stored locally"
* I think we should allow either passing the api_key param or just set the LANCEDB_API_KEY env var?

Reply:
@AyushExel  for last point - I am adding a logic to check if api_key is not provided then we check the OS.env variable , if thats not set then we raise an appropriate error.